src/Fixer/Basic/BracesFixer.php

```diff
@@ -124,33 +136,53 @@ public function bar($baz)
     /**
      * {@inheritdoc}
      *
-     * Must run before ArrayIndentationFixer, MethodArgumentSpaceFixer, MethodChainingIndentationFixer.
-     * Must run after ClassAttributesSeparationFixer, ElseifFixer, LineEndingFixer, MethodSeparationFixer, NoAlternativeSyntaxFixer, NoEmptyStatementFixer, NoUselessElseFixer, SingleTraitInsertPerStatementFixer.
+     * Must run before HeredocIndentationFixer.
+     * Must run after ClassAttributesSeparationFixer, MethodSeparationFixer, NoAlternativeSyntaxFixer, NoEmptyStatementFixer, NoUselessElseFixer, SingleTraitInsertPerStatementFixer.
      */
     public function getPriority()
     {
         return -25;
     }
 
-    /**
-     * {@inheritdoc}
-     */
-    public function isCandidate(Tokens $tokens)
+    public function configure(array $configuration = null)
     {
-        return true;
+        parent::configure($configuration);
+
+        $this->getCurlyBracesPositionFixer()->configure([
+            'control_structures_opening_brace' => $this->translatePositionOption($this->configuration['position_after_control_structures']),
+            'functions_opening_brace' => $this->translatePositionOption($this->configuration['position_after_functions_and_oop_constructs']),
+            'anonymous_functions_opening_brace' => $this->translatePositionOption($this->configuration['position_after_anonymous_constructs']),
+            'classes_opening_brace' => $this->translatePositionOption($this->configuration['position_after_functions_and_oop_constructs']),
+            'anonymous_classes_opening_brace' => $this->translatePositionOption($this->configuration['position_after_anonymous_constructs']),
+            'allow_single_line_empty_anonymous_classes' => false,
+            'allow_single_line_anonymous_functions' => $this->configuration['allow_single_line_closure'],
+        ]);
+
+        $this->getControlStructureContinuationFixer()->configure([
+            'keyword_position' => self::LINE_NEXT === $this->configuration['position_after_control_structures']
+                ? ControlStructureContinuationFixer::NEXT_LINE
+                : ControlStructureContinuationFixer::SAME_LINE,
+        ]);
+
+        $this->configuration = $configuration;
     }
 
     /**
      * {@inheritdoc}
      */
     protected function applyFix(\SplFileInfo $file, Tokens $tokens)
     {
-        $this->fixCommentBeforeBrace($tokens);
-        $this->fixMissingControlBraces($tokens);
-        $this->fixIndents($tokens);
-        $this->fixControlContinuationBraces($tokens);
-        $this->fixSpaceAroundToken($tokens);
-        $this->fixDoWhile($tokens);
+        for ($index = \count($tokens) - 1; $index > 0; --$index) {
```

Comment:
questions:
- why adding those spaces before running the proxy fixers?
- if someone would be using the proxied fixers directly, without BracesFixer, who would add those spaces?

Reply:
This is something I added as a TODO task in #4885. There is a PR (#4435) that could be a candidate replacement so I didn't implement one myself for now.

I moved the task here.

Reply:
nice!

Reply:
Just noticed I didn't actually add a TODO task to #4885 so I did it :smile: